### PR TITLE
Add the ability to configure GenericRemoteStorage for the scrubber

### PR DESCRIPTION
**Pull request by arpad-m**
Earlier work (#7547) has made the scrubber internally generic, but one could only configure it to use S3 storage.

This is the final piece to make (most of, snapshotting still requires S3) the scrubber be able to be configured via GenericRemoteStorage.

I.e. you can now set an env var like:

```
REMOTE_STORAGE_CONFIG='remote_storage = { bucket_name = "neon-dev-safekeeper-us-east-2d", bucket_region = "us-east-2" }
```

and the scrubber will read it instead.

### CODE REVIEW

**Comment by jcsp** [comment]
I can see the motivation for using a TOML input, although it's a bit more awkward to invoke.  It's probably fine if we have a clear example in the engineering book.  Pperhaps avoid people having to edit toml inline by providing some example for Azure like:
```
export ABS_CONTAINER=
export ABS_ACCOUNT=
export ABS_REGION=
export REMOTE_STORAGE_CONFIG=... the toml including the vars above

```

File: storage_scrubber/src/lib.rs [hunk 1]
```diff
@@ -230,13 +230,58 @@ pub fn remote_timeline_path_id(id: &TenantShardTimelineId) -> RemotePath {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(deny_unknown_fields)]
-pub struct BucketConfig {
+pub enum BucketConfig {
+    Legacy(BucketConfigLegacy),
```
Comment (jcsp) [inline storage_scrubber/src/lib.rs:234]: I'd prefer to eliminate this Legacy variant: could we just always have a regular RemoteStorageConfig, but for the S3-only code paths, extract the bucket+region from it and assert out if it's not an S3 config?

Comment (arpad-m) [reply storage_scrubber/src/lib.rs]: done

**Comment by arpad-m** [comment]
For azure, it would probably be something like:
```
export AZURE_CONTAINER=neon-dev-storage-eastus2
export AZURE_STORAGE_ACCOUNT=psdeveastus2neonbuild
export AZURE_REGION=eastus2
export REMOTE_STORAGE_CONFIG="remote_storage = { container_name = '${AZURE_CONTAINER}', storage_account = '${AZURE_STORAGE_ACCOUNT}', container_region = '${AZURE_REGION}' }"
```

**Review by jcsp — APPROVED** [review]
High level approach LGTM, will let Chi do a detailed review

**Review by skyzh — APPROVED** [review]

**Comment by arpad-m** [comment]
[e2e CI](https://github.com/neondatabase/cloud/actions/runs/11707740596) rerun had the error:
```
Error response from daemon: manifest for 369495373322.dkr.ecr.eu-central-1.amazonaws.com/neon:11707092750 not found: manifest unknown: Requested image not found
```

I suppose this is because the image is already deleted due to being too old. so I'm merging main into the branch again.

**Pull request merged**